frontend/src/pages/inventory/Inventory.tsx

```diff
@@ -1,66 +1,165 @@
-import { useState, useEffect } from 'react';
+import React, { useState, useEffect } from 'react';
 import { Link } from 'react-router-dom';
-import { Button } from '@material-ui/core';
+import {
+  Button,
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableRow,
+  IconButton,
+  Menu,
+  MenuItem,
+  InputBase
+} from '@material-ui/core';
+import { KeyboardArrowDown, KeyboardArrowUp, Search } from '@material-ui/icons/';
 
 import { Container, Card } from '../../components';
-import { API_GOOD } from '../../utils/api';
+import { API_ARCHIVE_GOOD, API_GOOD } from '../../utils/api';
 import ImportButton from './csv-impex/csv-import';
 import ExportButton from './csv-impex/csv-export';
 import './Inventory.scss';
 import { Item } from '../../interfaces/Items';
+import ItemRow from './item-row/ItemRow';
+import {
+  sortItemsByType,
+  filterForRawItems,
+  filterForSemiItems,
+  filterForFinishedItems
+} from './Filters';
 
 export default function Inventory() {
   const [items, setItems] = useState<Item[]>([]);
+  const [displayItems, setDisplayItems] = useState<Item[]>([]);
+  const [allOpen, setAllOpen] = useState<boolean>(false);
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+  const [filter, setFilter] = useState<number>(0);
+  const [search, setSearch] = useState<string>('');
```

Comment:
That's a lot of states you could merge into a single object state. Something like this is fine:

```js 
interface TableState {
  allOpen: boolean;
  filter: number;
  search: string;
} 

const [table, setTable] = useState<TableState>({
  allOpen: false,
  filter: 0,
  search: ''
});
```